src/main/java/com/chargerrobotics/subsystems/LimelightSubsystem.java

```diff
@@ -70,21 +103,150 @@ public double getV() {
 		return v;
 	}
 
-	// distance in inches
-	public double distance() {
-		if (v == 0) {
-			return -1.0;
+	public double getS() {
+		return s;
+	}
+
+	public double distM;
+	public double skewangle;
+	public double xcoord;
+	public double ycoord;
+	public double realangle;
+	public double compassdeg;
+
+	// cycles through pipelines looking for a target
+	public int findPL() {
+		for (int b = 0; b <= 10; b++) {
+			pip.setNumber(b);
+			pl = b;
+			if (v == 1)
+				break;
+		}
+		return pl;
+	}
+
+	int pl = findPL();
```

Comment:
@yakiddinmeb , what I meant by moving this was to put it within methods such as `distanceM()`. For instance, on line 131:
```java
public Double distanceM() {
    int pl = findPL();
    if (p1 == shooterTarget || p1 == fuelTarget)
    ...
}
```
This would also go into any other method that uses `pl`.